hadoop-ozone/integration-test-recon/src/test/java/org/apache/hadoop/ozone/recon/TestReconContainerEndpoint.java

```diff
@@ -107,6 +110,16 @@ public void testContainerEndpointForFSOLayout() throws Exception {
         recon.getReconServer().getOzoneManagerServiceProvider();
     impl.syncDataFromOM();
 
+    // Wait for async event processing to complete
+    // Events are processed asynchronously, so wait for processing to finish
+    ReconTaskControllerImpl reconTaskController =
+        (ReconTaskControllerImpl) recon.getReconServer().getReconTaskController();
+    CompletableFuture<Void> completableFuture =
+        omMetaManagerUtils.waitForEventBufferEmpty(reconTaskController.getEventBuffer());
+    while (!completableFuture.isDone()) {
```

Comment:
we can use GenericTestUtils.waitFor() for this case, instead of infinite loop. similar changes can be done other places